tla/Traceccfraft.tla

```diff
@@ -339,22 +332,13 @@ TraceStats ==
     TLCGet("stats")
 
 TraceMatched ==
-    LET d == TraceStats.diameter IN
-    d < Len(TraceLog) => Print(<<"Failed matching the trace " \o JsonFile \o " to (a prefix of) a behavior:", 
-                                    TraceLog[d+1], "TLA+ debugger breakpoint hit count " \o ToString(d+1)>>, FALSE)
-
-TraceStateSpace ==
-    \* TODO This can be removed when Traceccfraft is done.
-    /\ JsonFile \in KnownScenarios => TraceStats.distinct = Len(TraceLog)
+    TraceStats.diameter - 1 = Len(TraceLog)
 
 TraceAccepted ==
-    /\ TraceMatched
-    /\ TraceStateSpace
-
-TraceInv ==
-    \* This invariant may or may not hold depending on the level of non-determinism because
-     \* of holes in the log file.
-    TraceStats.distinct <= TraceStats.diameter
+    /\ PrintT("TraceStats.diameter " \o ToString(TraceStats.diameter) \o " TraceStats.distinct " \o ToString(TraceStats.distinct) \o " Len(TraceLog) " \o ToString(Len(TraceLog)))
+    /\
+        \/ TraceMatched
```

Comment:
`TraceMatched` cannot be an invariant because it is not true or false of a state.  Instead, it is true or false of a set of behaviors.

Reply:
Btw. I'm somewhat unsatisfied with the current error reporting.  Fundamentally, we face the problem that an error of trace validation is a set of behaviors and not just a single behavior --- TLC has only very limited and the VSCode extension has zero support to report such an error.  At least, TLC can now verify *and report a counterexample* for the following property:

```tla
TraceProp ==
    [](l <= Len(TraceLog) => [](TLCGet("queue") > 0 \/ l > Len(TraceLog)))
```

`TraceProp` is supposed to replace `TraceMatched` after a bit more testing.

Reply:
Ok, I'll make it a property.